platform/security/idp/security-idp-client/src/main/java/org/codice/ddf/security/idp/client/IdpHandler.java

```diff
@@ -606,12 +604,23 @@ private String encodeRedirectRequest(String request) throws WSSecurityException,
     return URLEncoder.encode(RestSecurity.deflateAndBase64Encode(request), "UTF-8");
   }
 
-  private String encodePostRequest(String request) throws WSSecurityException, IOException {
+  private String encodePostRequest(String request) throws WSSecurityException {
     return Base64.getEncoder().encodeToString(request.getBytes(StandardCharsets.UTF_8));
   }
 
   private String recreateFullRequestUrl(HttpServletRequest request) {
     StringBuffer requestURL = request.getRequestURL();
+
+    try {
+      URL url = new URL(requestURL.toString());
+      if (url.getHost().equals(SystemBaseUrl.EXTERNAL.getHost())
+          && String.valueOf(url.getPort()).equals(SystemBaseUrl.EXTERNAL.getPort())) {
+        requestURL = new StringBuffer(SystemBaseUrl.EXTERNAL.constructUrl(request.getRequestURI()));
+      }
+    } catch (MalformedURLException e) {
+      LOGGER.debug("Unable to convert request URL to URL object.");
```

Comment:
❓ Wouldn't SSO fail if this happens? Should this be logged as an error instead? (applies to the multiple places where you have this same logic)

Reply:
I'll change it to an error log